### ISSUE TITLE
Help improve isolation of specs

### ISSUE THREAD

**Issue by ciaranmcnulty** [phpspec/phpspec#630]
PhpSpec is not designed for functional testing, so it would make sense to try and dissuade this usage. 

It's hard to detect functional testing, one proxy metric might be to look at the number of objects involved in a spec using `get_declared_classes` and if it's above a certain threshold show some sort of non-failing warning.

e.g. "Warning: >100 objects were involved in this spec, it is likely to be fragile"

**Comment by stof** [comment]
@ciaranmcnulty get_declared_class won't help to detect this. It can only detect classes which are loaded during this spec, but your functional tests might involved only classes which are already defined (if there were speced themselves previously for instance). It does not allow measuring the number of involved classes.

**Comment by ciaranmcnulty** [comment]
True, it'd throw a warning on the first spec that included too much I guess?

This may be possible with Xdebug, either counting objects involved in the call or something more complex like checking whether they're in `src`
